playbooks/roles/logstash/tasks/main.yml

```diff
@@ -2,54 +2,64 @@
 # - oraclejdk
 # - elasticsearch
 ---
+- name: Add Elasticsearch apt key.
+  apt_key:
+    url: http://packages.elasticsearch.org/GPG-KEY-elasticsearch
+    state: present
 
-- name: Ensure app apt dependencies are installed
-  apt: pkg={{ item }} state=installed
-  with_items:
-    - redis-server
-
-- name: Ensure {{ logstash_app_dir }} exists
-  file: path={{ logstash_app_dir }} state=directory owner=root group=root mode=0755
-
-- name: Ensure subdirectories exist
-  file: path={{ logstash_app_dir }}/{{ item }} owner=root group=root mode=0755 state=directory
-  with_items:
-    - bin
-    - etc
-    - share
-
-- name: ensure logstash config is in place
-  template: src=logstash.conf.j2 dest={{ logstash_app_dir }}/etc/logstash.conf owner=root group=root mode=0644
-  notify: restart logstash
-
-- name: ensure logstash upstart job is in place
-  template: src=logstash.upstart.conf.j2 dest=/etc/init/logstash.conf owner=root group=root mode=0755
+- name: Add Logstash repository.
+  apt_repository:
+    repo: 'deb http://packages.elasticsearch.org/logstash/2.3/debian stable main'
+    state: present
 
 - name: ensure logstash has a logging dir at {{ logstash_log_dir }}
-  file: path={{ logstash_log_dir }} owner=root group=root mode=0755 state=directory
-
-- name: ensure we have the specified logstash release
-  get_url: url={{ logstash_url }} dest={{ logstash_app_dir }}/share/{{ logstash_file }}
-
-- name: ensure symlink with no version exists at {{ logstash_app_dir }}/share/logstash.jar
-  file: src={{ logstash_app_dir }}/share/{{ logstash_file }} dest={{ logstash_app_dir }}/share/logstash.jar state=link
-
-- name: start logstash
-  action: service name=logstash state=started enabled=yes
-
-- name: Ensure we are running
-  wait_for: port={{ logstash_syslog_port }} host=localhost timeout=60
-
-- name: Copy logstash es index template
-  copy: src=template_logstash.json dest=/etc/elasticsearch/template_logstash.json
-
-- name: Enable logstash es index template
-  shell: chdir=/etc/elasticsearch executable=/bin/bash curl -XPUT 'http://localhost:9200/_template/template_logstash' -d @template_logstash.json
-
+  file: path={{ logstash_log_dir }} owner=root group=root mode=0775 state=directory
+
+- name: ensure logstash has a configure dir at {{ logstash_app_dir }}
+  file: path={{ logstash_app_dir }}/etc owner=root group=root mode=0755 state=directory
```

Comment:
There was a line above that did this (and other directories), `Ensure subdirectories exist`.

Why not just use that? If we don't need all of the subdirectories, we can remove them from the `with_items` clause, but let's not change any more than we need to.